native/src/main/java/io/ballerina/lib/data/xmldata/xml/XmlParser.java

```diff
@@ -892,26 +895,31 @@ private Map<QualifiedName, Field> getAllFieldsInRecordType(RecordType recordType
         Map<String, List<QualifiedName>> fieldNames = new HashMap<>();
         Map<String, Field> recordFields = recordType.getFields();
         for (String key : recordFields.keySet()) {
+            QualifiedNameMap<Field> attributeMap = xmlParserData.attributeHierarchy.peek();
             QualifiedName modifiedQName =
                     modifiedNames.getOrDefault(key, new QualifiedName(Constants.NS_ANNOT_NOT_DEFINED, key, ""));
             String localName = modifiedQName.getLocalPart();
-            if (fieldMap.containsKey(modifiedQName)) {
+            if (!attributeMap.contains(modifiedQName)) {
+                fieldMap.put(modifiedQName, recordFields.get(key));
+                fieldNames.put(localName, new ArrayList<>(List.of(modifiedQName)));
+            } else if (attributeMap.contains(modifiedQName) && modifiedQName.getAttributeState() == NOT_DEFINED) {
+                if (!key.equals(attributeMap.get(modifiedQName).getFieldName())) {
+                    modifiedQName.setAttributeState(ELEMENT);
+                    fieldMap.put(modifiedQName, recordFields.get(key));
+                    fieldNames.put(localName, new ArrayList<>(List.of(modifiedQName)));
```

Comment:
`List.of(modifiedQName)`
This can be define as a variable in here

Reply:
This will return an immutable list. I have to modify `modifiedQName` and then create this.